### ISSUE TITLE
Checkbox set w/open text field functionality not matching accessibility best practices

### ISSUE THREAD

**Issue by cgednalske** [department-of-veterans-affairs/vets-design-system-documentation#604]
# Bug Report

- [ ] I’ve searched for any related issues and avoided creating a duplicate issue.

## What happened

Design and Accessibility have advised that open text fields associated with a list of checkboxes should be visible by default. The checkbox set components in the design system do not function in this manner, with each open text field currently hidden until the corresponding button is selected. These components should be updated to function according to best practices. Screenshot attached below.

<img width="563" alt="Checkboxes_text input field" src="https://user-images.githubusercontent.com/79372956/146837930-e32cbbae-07da-4c56-b15b-bd362647f0b1.png">

## What I expected to happen

We expected the text input field to be always be visible based on best practices for accessibility. Our FE dev Adam Whitlock discovered this difference when coding the checkbox set and found the existing system pattern did not allow for an already-visible text input field. 

We confirmed with Accessibility Specialists Angela Fowler and Josh Kim that the text input field should always be visible in this case. If it only appears when other is selected, that would mean the screen changes without the user knowingly doing something to initiate that change.

**Comment by pdavies88** [comment]
Related Ticket https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/605

Currently in staging when logged in and going to `/profile` and using the Edit button to open the Pronouns options the text field is being shown by default
![Screen Shot 2022-01-04 at 11.15.43 AM.png](https://images.zenhubusercontent.com/618570fbb420f53ff83732e5/3b1c719d-1ace-4deb-bfc3-40c637fb2cf1)

**Comment by MarciMcGuireGCIO** [comment]
Updating this ticket to note that we met with several members of the Design System team and Angela Fowler on 1/6/2022 to discuss options for this and a similar challenge with radio button groups.  

The consensus was that we should have the associated text field for entering an "other" option visible by default, but that we should also provide a visual cue that it is linked with the "other" option.    That would be accomplished by following a pattern similar to what is already shown in Design System for a [Radio Button Group with expanded content](https://design.va.gov/components/form-controls#radio-buttons) where a small blue line connects the radio button to the content below it.  

[This pattern has now been mocked up in Sketch](https://www.sketch.com/s/ba254d92-3c3d-4eba-825d-d7f5bda35565/a/nRMa9qd) and is also on Staging so that it can be looked at interactively.  In order to view in Staging, log in with vets.gov.user+36@gmail.com, access Profile and refer to the sections for Pronouns.

**Comment by humancompanion-usds** [comment]
I'm just going to update our pattern with the current example, which doesn't use an exposed field at all and then close this.

**Comment by humancompanion-usds** [comment]
@MarciMcGuireGCIO - I know this was filed 9 months ago so didn't expect the same user account to work and that's turned out to be true. I'm having a hard time finding the Pronouns mentioned above in staging. Is there a user I can use to login and see it? I jumped to updating Gender where we don't expose an additional field. I'm happy to have updated screenshots in that pattern but realizing that, duh, this is Pronouns, not Gender. Wondering if we do the same thing there or something different.

**Comment by cgednalske** [comment]
Hi @humancompanion-usds! I can answer that as I'm still working on this Profile section. Pronouns mentioned above is not in staging because we've only been able to add Gender identity so far due to waiting on PRA approval. Pronouns and Sexual orientation (which both have the text input field with checkboxes) are on deck so to speak, but they have changed slightly since then. There was a ton of discussion around this with the design council at the time, resulting in removing the blue line and adding more space before the text input field (image below). Here are Sketch files for both: [Pronouns](https://www.sketch.com/s/ba254d92-3c3d-4eba-825d-d7f5bda35565/a/ygWOEPy) and [Sexual orientation](https://www.sketch.com/s/ba254d92-3c3d-4eba-825d-d7f5bda35565/a/8yoERJO). 

Additionally, you might have noticed that Gender identity has radio buttons, not checkboxes. That was due to a backend constraint; we had to stick with radio buttons though our original intention was to use checkboxes like the other fields. 

<img width="618" alt="Screen Shot 2022-09-08 at 2 38 22 PM" src="https://user-images.githubusercontent.com/79372956/189230721-f00ad896-42f2-4876-929b-1f944c3f3590.png">

**Comment by cgednalske** [comment]
@adamwhitlock1 is there a way for Matt to see Pronouns and Sexual orientation on staging right now using a specific account with that feature flag? I'm guessing probably not, but it's worth a shot 😃

**Comment by adamwhitlock1** [comment]
Yes I can turn on the POC Pronouns and Sexual Orientation fields for @humancompanion-usds ! Since user 36 is kind of our go to for testing specific other features, lets turn it on for a different user.

I have gone ahead and added the feature toggle for just user **vets.gov.user+75@gmail.com** so that those fields can be seen.

The state of the UI is as follows:

- Pronouns still has the blue border (simply because we hadn't changed that yet, and instead just hid them all for the time being)
- Sexual Orientation appears more as we had envisioned it without the blue border on the side


Let me know if you have any more questions. 😃

**Comment by humancompanion-usds** [comment]
Thank you @cgednalske and @adamwhitlock1! I'll have a look.

@cgednalske - Where do we plan on using pronouns? How do we address a Veteran if they select multiple pronouns? Also, do you recall when this came to DSC? I'd like to find the notes from the discussion that day to be sure I'm addressing anything that was raised in the guidance I'm writing. Thanks!

**Comment by cgednalske** [comment]
Thanks @adamwhitlock1! That's perfect. 🎉 

@humancompanion-usds - the Sketch files still reflect the latest design, but it sounds like staging is really close. 👍 

> Where do we plan on using pronouns?

Pronouns (and the other similar fields on the Personal info page) are primarily accessed by VHA, so will be used mostly when Veterans visit VA facilities and speak to those VA representatives. Previously there wasn't a way to convey these preferences ahead of time, which was creating distressing encounters for both Veterans and VA staff. 

The info can also be accessed by non-health related areas like VBA and NCA, so it could be used elsewhere as well. In those cases, who has access vs. who will actually _use_ the information is a little trickier to pinpoint, but definitely VA facilities and representatives. To be transparent about privacy, we're telling Veterans that VA staff outside of the health care system may be able to view this information.

> How do we address a Veteran if they select multiple pronouns?

If there are multiple selections for Pronouns then VA folks can address Veterans using either selection. Though probably an overall healthcare question first and foremost, Gender identity might be trickier in that regard if we ever switch to checkboxes for that field.

> Also, do you recall when this came to DSC? I'd like to find the notes from the discussion that day to be sure I'm addressing anything that was raised in the guidance I'm writing.`

It was a while back, but I will hunt for this info and get back to you!

**Comment by humancompanion-usds** [comment]
Okay, I've update the [Gender pattern](https://design.va.gov/patterns/ask-users-for/gender) and created a [Pronouns pattern](https://design.va.gov/patterns/ask-users-for/pronouns). @cgednalske and @davidakennedy - I would appreciate you giving those patterns a review and providing me with any feedback you may have. Thanks!

**Comment by cgednalske** [comment]
@humancompanion-usds will take a look! And it looks like the Design Council meeting for this was on January 21st, but that's all the information I've been able to find. Hope it is helpful!

**Comment by cgednalske** [comment]
@humancompanion-usds Everything looks good to me 👍  We're currently working on changing a small part of the wording in the Gender identity expanded FAQ info (replacing the mention of the VBA and NCA with a more general description). Is that something you'd like to update again once we do?

**Comment by humancompanion-usds** [comment]
Thanks @cgednalske. Yes, send me a screenshot when it changes and I'll make the update.
